examples/tensorflow_hello_world/helloworld.py

```diff
@@ -1,11 +1,211 @@
+        model_provider=model_provider,
+        xy_validation=xy_validation,
+    )
+
+    # printing some basic statistics
+    print_dataset_stats(xy_train_partitions, xy_validation, xy_test)
+
+    # calling fit on the coordinator starts the training
+    # and print validation metrics after each global round
+    validation_metrics, _, _, _ = coordinator.fit(num_rounds=R)
+    print_validation_metrics(validation_metrics)
+
+    # evaluate the model with the test set
+    loss, accuracy = coordinator.evaluate(xy_test)
+    print(f"---\nTest completed!\nLoss: {loss} | Test accuracy: {accuracy}")
+
+
+# specify learning rate function, in this case with exponential decay
+def learning_rate_fn(epoch_base, lr_initial=DEFAULT_LR, k=DEFAULT_K):
+    def exp_decay(epoch_optimizer: int) -> float:
+        epoch = epoch_base + epoch_optimizer
+        return lr_initial * math.exp(-k * epoch)
+
+    return exp_decay
+
+
+# this function contains the model architecture
+def create_and_compile_model(epoch_base=0):
+    def add_convolution(filters, kernel_inizializer):
+        convolution = Conv2D(
+            filters,
+            kernel_size=(5, 5),
+            strides=(1, 1),
+            kernel_initializer=kernel_inizializer,
+            padding="same",
+            activation="relu",
+        )
+        return convolution
+
+    ki = tf.keras.initializers.glorot_uniform(seed=42)
+
+    # Architecture
+    inputs = Input(shape=INPUT_SHAPE)
+    x = add_convolution(filters=32, kernel_inizializer=ki)(inputs)
+    x = MaxPool2D(pool_size=(2, 2), strides=(2, 2))(x)
+    x = add_convolution(filters=64, kernel_inizializer=ki)(x)
+    x = MaxPool2D(pool_size=(2, 2), strides=(2, 2))(x)
+    x = Flatten()(x)
+    x = Dense(512, kernel_initializer=ki, activation="relu")(x)
+    outputs = Dense(NUM_CLASSES, kernel_initializer=ki, activation="softmax")(x)
+
+    model = tf.keras.Model(inputs=inputs, outputs=outputs)
+
+    # Compile model with exponential learning rate decay
+    lr_fn = learning_rate_fn(epoch_base=epoch_base)
+    optimizer = tf.keras.optimizers.Adam(lr=lr_fn(0))
+
+    model.compile(
+        loss=tf.keras.losses.categorical_crossentropy,
+        optimizer=optimizer,
+        metrics=["accuracy"],
+    )
+    return model
+
+
+# iterate through each partition (a client's training data)
+# and init a Participant class
+def init_participants(xy_train_partitions, model_provider, xy_validation):
+    potential_participants = []
+    for client_id, xy_train in enumerate(xy_train_partitions):
+        participant = Participant(
+            cid=client_id,
+            model_provider=model_provider,
+            xy_train=xy_train,
+            xy_val=xy_validation,
+            num_classes=NUM_CLASSES,
+            batch_size=B,
+        )
+        potential_participants.append(participant)
+    return potential_participants
+
+
+# init the Coordinator
+def init_coordinator(participants, xy_train_partitions, model_provider, xy_validation):
```

Comment:
I'd prefer to remove the `xy_train_partitions` argument since it's only being used for assertion - we should expect the `participants` argument to be of valid length.